public/class-award-force-sso.php

```diff
@@ -57,15 +57,21 @@ private function getSlug(WP_User $user)
      */
     private function requestSlug(WP_User $user)
     {
-        $response = $this->api->post('/user', [
-            'form_params' => [
-                'email'     => $user->user_email,
-                'firstName' => $user->user_firstname ?: 'First',
-                'lastName'  => $user->user_lastname ?: 'Last'
-            ]
-        ]);
-
-        return $response->slug;
+        try {
+            $response = $this->api->post('/user', [
+                'email' => $user->user_email,
+                'first_name' => $user->user_firstname ?: 'First',
+                'last_name' => $user->user_lastname ?: 'Last',
+                'password' => uniqid(),
+            ]);
+            return $response->slug;
+        } catch (Exception $e) {
+            if ($e->getCode() === AwardForceAPIV2::$emailAlreadyExists) {
+                $response = $this->api->get("user/".$user->user_email);
+                return $response->slug;
+            }
+            $this->api->handleException($e);
+        }
```

Comment:
How about:
```
if ($response = $this->api->get("user/".$user->user_email)) {
    return $response->slug;
}

$response = $this->api->post('/user', [
  'email' => $user->user_email,
  'first_name' => $user->user_firstname ?: 'First',
  'last_name' => $user->user_lastname ?: 'Last',
  'password' => uniqid(),
]);

return $response->slug;
```

Reply:
yep, the issue here is the 422 error code is managed in the client as an error and it throws an exception, so the check ìf ($response. ...)` doesn't do the job. Anyway, I realised there was a something wrong since we are returning the slug into the Location header after user creation as part of the user url, so I changed a bit the way it's handled here.